src/bodywork/k8s/secrets.py

```diff
@@ -162,25 +189,32 @@ def delete_secret(namespace: str, name: str) -> None:
     k8s.CoreV1Api().delete_namespaced_secret(namespace=namespace, name=name)
 
 
-def list_secrets(namespace: str, group: str = None) -> Dict[str, Dict[str, str]]:
+def list_secrets(namespace: str, group: str = None) -> Dict[str, Secret]:
     """Get all secrets and their (decoded) data.
 
     :param namespace: Namespace in which to list secrets.
     :param group: Group of secrets to list.
 
     """
-    secrets = k8s.CoreV1Api().list_namespaced_secret(
-        namespace=namespace,
-        label_selector=f"{SECRET_GROUP_LABEL}={group}",
-    )
-    secret_data_base64 = {
-        s.metadata.name: s.string_data if s.string_data else s.data
-        for s in secrets.items
-    }
-    secret_data_decoded = {
-        secret_name: {
-            key: b64decode(value).decode("utf-8") for key, value in secret_data.items()
-        }
-        for secret_name, secret_data in secret_data_base64.items()
+    if group is None:
+        result = k8s.CoreV1Api().list_namespaced_secret(namespace=namespace)
+    else:
+        result = k8s.CoreV1Api().list_namespaced_secret(
+            namespace=namespace,
+            label_selector=f"{SECRET_GROUP_LABEL}={group}",
+        )
+    secrets = {
+        s.metadata.name: Secret(
+            s.metadata.name,
+            s.metadata.labels[SECRET_GROUP_LABEL]
+            if s.metadata.labels and SECRET_GROUP_LABEL in s.metadata.labels
+            else None,
+            s.string_data if s.string_data else s.data,
+        )
+        for s in result.items
     }
-    return secret_data_decoded
+
+    for key, value in secrets.items():
```

Comment:
Can I suggest that you roll this into the creation of the `secrets` dict or add the logic into the constructor for the `Secrets` class?

Reply:
I disagree here, this loop contains a line to decode to UTF8. You would end up restricting the creation of the Secrets Dataclass just for Secrets obtained directly from the API i.e. just from this method. 

A DTO/dataclass should not impose that restriction, the data needs to be able to flow the other way.

Reply:
What I'm taking exception to here, is mutating the `Secrets` object (the `data` field) from the outside. This doesn't feel right to me. I'd rather there was a `data_decoded` field so this wasn't necessary. Have I understood this correctly?

Reply:
I understand where you're coming from but I think it's unnecessary because when we create or update a new secret we do not need to encode it to be able to save/update, that is handled by Kubernetes so we only ever use plain text.  

This object is our simplified abstraction of a kubernetes Secret. I cannot see the value add of keeping the encoded data in the object in addition to the decoded data, I cannot see a potential use for that.

In the previous implementation we returned just the decoded data in the dictionary so there is no change in that respect either.

Maybe it's easier to discuss this one in person.

Reply:
Yeah, let's chat about this.

Reply:
The other option is to decode elsewhere - i.e. wherever it needs to be decoded, e.g. in the CLI function for printing to stdout, etc.

Reply:
It's now decoded in the constructor, we should have done this in the first place instead of looping through the values afterwards. 

Also the previous implementations would decode the string_data field if that was returned. This would result in an error if that ever happened because the string_data field is just for decoded data i.e. plain text.

Reply:
Nice one - looks good.

Have you blacked that file? I'm curious if that block is kosher or not.

Reply:
Yeah it's Black that formatted it that way. It does look a bit weird to me as well, I don't like how the `else None` is on a separate line when there is space to fit it on the same line.